superset/db_engine_specs.py

```diff
@@ -74,6 +84,115 @@ def extra_table_metadata(cls, database, table_name, schema_name):
+                                 header=header,
+                                 names=names,
+                                 index_col=index_col,
+                                 squeeze=squeeze,
+                                 prefix=prefix,
+                                 mangle_dupe_cols=mangle_dupe_cols,
+                                 skipinitialspace=skipinitialspace,
+                                 skiprows=skiprows,
+                                 nrows=nrows,
+                                 skip_blank_lines=skip_blank_lines,
+                                 parse_dates=parse_dates,
+                                 infer_datetime_format=infer_datetime_format,
+                                 dayfirst=dayfirst,
+                                 thousands=thousands,
+                                 decimal=decimal,
+                                 quotechar=quotechar,
+                                 escapechar=escapechar,
+                                 comment=comment,
+                                 encoding='utf-8',
+                                 error_bad_lines=error_bad_lines,
+                                 chunksize=chunksize,
+                                 iterator=True)
+        df = pandas.DataFrame()
+        df = pandas.concat(chunk for chunk in chunks)
+        return df
+
+    @staticmethod
+    def df_to_db(df, name, con, schema, if_exists, index,
+                 index_label, chunksize):
+        engine = create_engine(con, echo=False)
+        # Use Pandas to parse dataframe to database
+        df.to_sql(name=name, con=engine, schema=schema, if_exists=if_exists,
+                  index=index, index_label=index_label, chunksize=chunksize)
+
+        database = (
+            db.session
+                .query(type(query)) #Database
+                .filter_by(sqlalchemy_uri=con)
+                .first()
+        )
+        table.database_id = database.id
+        table.user_id = g.user.id
+        table.database = database
+        table.schema = form.schema.data
+        db.session.add(table)
+        db.session.commit()
+
+        #post_add
+        table.fetch_metadata()
+        security.merge_perm(sm, 'datasource_access', table.get_perm())
+        if table.schema:
+            security.merge_perm(sm, 'schema_access', table.schema_perm)
+
+
+    @staticmethod
+    def upload_csv(form, table):
+        def allowed_file(filename):
+            # Only allow specific file extensions as specified in the config
+            extension = os.path.splitext(filename)[1]
+            return extension and extension[1:] in config['ALLOWED_EXTENSIONS']
+
+        filename = secure_filename(form.csv_file.data.filename)
+        if not allowed_file(filename):
+            flash("Invalid file trype selected.", 'alert')
+            return False
+        df = BaseEngineSpec.csv_to_df(names=form.names.data,
+                            filepath_or_buffer=filename,
```

Comment:
NIT: doesn't the linter trigger on this? `names` should be aligned with `filepath_or_buffer`, personally I'd break after the `(` and go with just 4 indents.

This is also a lot of boilerplate... There's a way to do this where the list of args is defined only once, in an array. We can chat about it.